src/merkle/partial_mt/mod.rs

```diff
@@ -234,46 +258,45 @@ impl PartialMerkleTree {
     }
 
     /// Updates value of the leaf at the specified index returning the old leaf value.
+    /// By default the specified index is assumed to belong to the deepest layer. If the considered
+    /// node does not belong to the tree, the first node on the way to the root will be changed.
```

Comment:
minor nit: I'd add a line break after the first line - the doc formatter would be able to process it better this way.

Also, is there a reason we are no longer returning the old value from here?

Reply:
I decided not to return the old value for the sake of uniformity -- neither `MerkleTree` nor `MerklePathSet` return an old value from `update_leaf()`, this return value is not used anywhere. But I can return it, maybe we will use this feature in future.

Reply:
Yes, let's return the old value form here. We might do the same in other places too later on.